modules/dnn/test/test_layers.cpp

```diff
@@ -1937,6 +1937,171 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_Test_Eltwise_unequal, Combine(
     dnnBackendsAndTargets()
 ));
 
+typedef testing::TestWithParam<tuple<Backend, Target> > Layer_Test_Eltwise_bcast;
+TEST_P(Layer_Test_Eltwise_bcast, change_input)
+{
+    // config
+    vector<string> ops_list = {"sum"};
+    vector<int> dims_list = {1, 2, 3, 4, 5};
+
+    struct util
+    {
+        // give n to generate all n-D arrays with 0 or 1
+        static void get_all_arr(vector<vector<int>> &arr, int n)
+        {
+            int total = 1 << n;
+            arr.assign(total, vector<int>(n, -1));
+            for (int i = 0; i < total; i++)
+                for (int j = 0; j < n; j++)
+                    arr[i][j] = (i >> (n - j - 1)) & 1;
+        }
+
+        // zero will replace all 0, one will replace all 1
+        static void replace(vector<vector<int>> &arr, int zero, int one)
+        {
+            for (int i = 0; i < arr.size(); i++)
+                for (int j = 0; j < arr[0].size(); j++)
+                    arr[i][j] = arr[i][j] ? one : zero;
+        }
+
+        // test if the shape can be forwarded
+        static int
+        test_bcast(const vector<int> &a_shape, const vector<int> &b_shape, const String &op,
+                int backend, int target)
+        {
+            Mat a = Mat::zeros((int) a_shape.size(), a_shape.data(), CV_32FC1);
+            Mat b = Mat::ones((int) b_shape.size(), b_shape.data(), CV_32FC1);
+
+            Net net;
+            LayerParams lp;
+            lp.type = "NaryEltwise";
+            lp.name = "testLayer";
+            lp.set("operation", op);
+            int id = net.addLayerToPrev(lp.name, lp.type, lp);
+            net.connect(0, 1, id, 1);
+
+            vector<String> inpNames(2);
+            inpNames[0] = "a";
+            inpNames[1] = "b";
+            net.setInputsNames(inpNames);
+            net.setInput(a, inpNames[0]);
+            net.setInput(b, inpNames[1]);
+
+            net.setPreferableBackend(backend);
+            net.setPreferableTarget(target);
+            try
+            {
+                Mat re = net.forward();
+                auto ptr_re = (float *) re.data;
+                for (int i = 0; i < re.total(); i++)
+                    if (op == "sum" && ptr_re[i] != 1)
+                        return -2; // sum result is wrong
+                return 1; // all right
+            } catch (std::runtime_error &e)
+            {
+                if ((string) e.what() == "fallback")
+                    return -3; // fallback to cpu
+                else
+                    return -4; // other error
+            }
+            catch (...)
+            {
+                return -1; // runtime error
```

Comment:
Can we use `ASSERT_*` instead right here?

Reply:
I have changed them to `ASSERT_*`